src/robotFactory.js

```diff
@@ -1,10 +1,78 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.coords = {};
 
-class FlyingRobot {}
+    this.coords.x = coords.x
+      ? coords.x
+      : 0;
```

Comment:
```suggestion
    this.coords = {
      x,
      y,
    };
```
this would be better approach here :)